src/private/m2tUpdater.m

```diff
@@ -27,7 +27,115 @@
 %   ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 %   POSSIBILITY OF SUCH DAMAGE.
 % =========================================================================
-  
+  fileExchangeUrl = about.website;
+  version = about.version;
+
+  mostRecentVersion = determineLatestRelease(version, fileExchangeUrl);
+  if askToUpgrade(mostRecentVersion, version, verbose)
+      tryToUpgrade(fileExchangeUrl, verbose);
+      userInfo(verbose, '');
+  end
+end
+% ==============================================================================
+function shouldUpgrade = askToUpgrade(mostRecentVersion, version, verbose)
+  shouldUpgrade = false;
+  if ~isempty(mostRecentVersion)
+      userInfo(verbose, '**********************************************\n');
+      userInfo(verbose, 'New version (%s) available!\n', mostRecentVersion);
+      userInfo(verbose, '**********************************************\n');
+
+      warnAboutUpgradeImplications(version, mostRecentVersion, verbose);
+      askToShowChangelog(version);
+      reply = input(' *** Would you like to upgrade? y/n [n]:','s');
+      shouldUpgrade = strcmpi(reply(1),'y');
```

Comment:
Here you need to check for an empty response, otherwise this fails. (User directly presses enter.)